drivers/ethernet/eth_stm32_hal.c

```diff
@@ -68,10 +136,41 @@ static void enable_canbus_eth_translator_filter(ETH_HandleTypeDef *heth,
 }
 #endif /*CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR*/
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
+						uint32_t PHYAddr,
+						uint32_t PHYReg,
+						uint32_t *RegVal)
+{
+	return HAL_ETH_ReadPHYRegister(heth, PHYAddr, PHYReg, RegVal);
+}
+#else
+static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
+						uint32_t PHYAddr,
+						uint32_t PHYReg,
+						uint32_t *RegVal)
+{
+	ARG_UNUSED(PHYAddr);
+	return HAL_ETH_ReadPHYRegister(heth, PHYReg, RegVal);
+}
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
```

Comment:
Putting `#ifdef` inside function would enhance readability.

Reply:
Should then all `#if defined(CONFIG_SOC_SERIES_STM32H7X)` be replaced by `#ifdef CONFIG_SOC_SERIES_STM32H7X` ?

Reply:
Sorry, I was not clear enough, you should be able to do something like:
```
static HAL_StatusTypeDef read_eth_phy_register(ETH_HandleTypeDef *heth,
						uint32_t PHYAddr,
						uint32_t PHYReg,
						uint32_t *RegVal)
{
#if defined(CONFIG_SOC_SERIES_STM32H7X)
	return HAL_ETH_ReadPHYRegister(heth, PHYAddr, PHYReg, RegVal);
#else
	ARG_UNUSED(PHYAddr);
	return HAL_ETH_ReadPHYRegister(heth, PHYReg, RegVal);
#endif
}
```

Reply:
Will be done within next push.